core/src/main/kotlin/knote/api/Page.kt

```diff
@@ -1,21 +1,27 @@
 package knote.api
 
 import knote.script.PageScript
+import knote.util.KObservableList
+import knote.util.KObservableMap
 import knote.util.KObservableObject
 import java.io.File
+import java.nio.file.Path
 import kotlin.script.experimental.api.ScriptDiagnostic
 
 interface Page {
     val id: String
     val fileObject: KObservableObject<Page, File>
     val fileContentObject: KObservableObject<Page, String>
+    val textObject: KObservableObject<Page, String>
     val compiledScriptObject: KObservableObject<Page, PageScript?>
     val reportsObject: KObservableObject<Page, List<ScriptDiagnostic>?>
     val resultObject: KObservableObject<Page, Any?>
```

Comment:
It may be helpful at the interfaces to start adding commentary as to what these objects are.  A  lot of them are seriously self-explanatory, but I'd like to see comments added for.  I know what these are now, but others looking at this for the first time may not. I'll add suggestions for comments I think could use an explanation and you're welcome to add them or correct them as needed.

- `fileContentObject`: (actually the first two are new to me maybe I'll edit if I can figure it out but these two would be nice)
- `textObject`:
- `reportsObject`: logging for a page
- `resultObject`: evaluation result printed in the console for a page
- `fileInputs`: (?) parameters for given script functions in a page. 

On a personal note I'm interested in looking at how `fileInputs` are determined.